code.test.js

```diff
@@ -6,19 +6,19 @@ eval(fs.readFileSync('code.js')+'');
 var graph = {'foo': {'boo': 7},
     'boo': {'foo': 3, 'bar': 2},
     'bar': {'boo': 4}};
-assert(JSON.stringify(augmentingPath(graph, 'foo', 'bar')) == JSON.stringify(['foo', 'boo', 'bar']));
+assert(JSON.stringify(augmentingPath(graph, 'foo', 'bar', visited = [])) == JSON.stringify(['foo', 'boo', 'bar']));
```

Comment:
Please don't change the prototype of the function (and in this case you shouldn't need to).